docs/source/tools/trigger-service.rst

```diff
@@ -29,75 +29,173 @@ Although as we'll see, the Trigger Service exposes an endpoint for end-users to
 
-.. code-block:: bash
+HTTP Request
+============
 
-   $curl \
-      -X POST localhost:8088/v1/start \
-      -H "Content-type: application/json" -H "Accept: application/json" \
-      -d '{"triggerName":"312094804c1468e2166bae3c9ba8b5cc0d285e31356304a2e9b0ac549df59d14:TestTrigger:trigger", "party": "alice"}'
-   {"result":{"triggerId":"4d539e9c-b962-4762-be71-40a5c97a47a6"},"status":200}
+- URL: ``/v1/triggers``
+- Method: ``POST``
+- Content-Type: ``application/json``
+- Content:
 
-``stop``
-********
+.. code-block:: json
 
-Stop a running trigger. Alice stops her running trigger like so.
+    {
+      "triggerName": "312094804c1468e2166bae3c9ba8b5cc0d285e31356304a2e9b0ac549df59d14:TestTrigger:trigger",
+      "party": "alice"
+    }
 
-.. code-block:: bash
+HTTP Response
+=============
+
+.. code-block:: json
+
+    {
+      "result":{"triggerId":"4d539e9c-b962-4762-be71-40a5c97a47a6"},
+      "status":200
+    }
 
-   $curl \
-      -X DELETE localhost:8088/v1/stop/4d539e9c-b962-4762-be71-40a5c97a47a6 \
-      -H "Content-type: application/json" -H "Accept: application/json"
-   {"result":{"triggerId":"4d539e9c-b962-4762-be71-40a5c97a47a6"},"status":200}
 
-``upload_dar``
+Stop a trigger
 **************
 
-Upload an automation DAR. If successful, the DAR's "main package ID" will be in the response (the main package ID for a DAR can also be obtained using ``daml damlc inspect-dar path/to/dar``).
+Stop a running trigger. Alice stops her running trigger like so.
 
-.. code-block:: bash
+HTTP Request
+============
 
-   $ curl -F 'dar=@/home/alice/test-model.dar' localhost:8088/v1/upload_dar
-   {"result":{"mainPackageId":"312094804c1468e2166bae3c9ba8b5cc0d285e31356304a2e9b0ac549df59d14"},"status":200}
+- URL: ``/v1/triggers/:id``
+- Method: ``DELETE``
+- Content-Type: ``application/json``
+- Content:
 
-``list``
-********
+HTTP Response
+=============
 
-List the DARS running on behalf of a given party. Alice can check on her running triggers as follows.
+- Content-Type: ``application/json``
+- Content:
 
-.. code-block:: bash
+.. code-block:: json
+
+   {
+     "result": {"triggerId":"4d539e9c-b962-4762-be71-40a5c97a47a6"},
+     "status":200
+   }
+
+List running triggers
+*********************
 
-   $curl \
-       -X GET localhost:8088/v1/list \
-       -H "Content-type: application/json" -H "Accept: application/json"
-       -d '{"party": "alice"}'
-   {"result":{"triggerIds":["4d539e9c-b962-4762-be71-40a5c97a47a6"],"status":200}
+List the Triggers running on behalf of a given party.
 
-``status``
-**********
+HTTP Request
+============
+
+- URL: ``/v1/triggers``
+- Method: ``GET``
+- Content-Type: ``application/json``
+- Content:
+
+.. code-block:: json
+
+    {"party": "alice"}
```

Comment:
Maybe we could use this opportunity to avoid a body on GET? AFAIK this can cause some issues. There's some related discussion on SO [here](https://stackoverflow.com/questions/978061/http-get-with-request-body). TL;DR it's not strictly ruled out by the spec but there used to be wording against it and some proxies and such might drop the body or raise an error.

We could use a query parameter instead.

Reply:
Ugh, that’s bad I didn’t even notice that. I’ve changed it to a query parameter now. Not quite happy with the fact that this makes this endpoint special so open to suggestions for improving this. I guess we could use query parameters everywhere but that’s also a bit annoying :thinking: